mmv1/third_party/terraform/services/apphub/data_source_apphub_discovered_service.go

```diff
@@ -0,0 +1,185 @@
+                                                        Elem: &schema.Resource{
+                                                            Schema: map[string]*schema.Schema{
+                                                                "gcp_project":{
+                                                                    Type: schema.TypeString,
+                                                                    Computed: true,
+                                                                },
+                                                                "location":{
+                                                                    Type: schema.TypeString,
+                                                                    Computed: true,
+                                                                },
+                                                                "zone":{
+                                                                    Type: schema.TypeString,
+                                                                    Computed: true,
+                                                                },
+                                                            },
+                                                        },        
+                                                },
+                                        },
+                                },
+                        },
+        },
+    }
+}
+
+func dataSourceApphubDiscoveredServiceRead(d *schema.ResourceData, meta interface{}) error {
+        config := meta.(*transport_tpg.Config)
+        userAgent, err := tpgresource.GenerateUserAgentString(d, config.UserAgent)
+        if err != nil {
+                return err
+        }
+
+        url, err := tpgresource.ReplaceVars(d, config, "{{ApphubBasePath}}projects/{{project}}/locations/{{location}}/discoveredServices:lookup?uri={{service_uri}}")
+        if err != nil {
+                return err
+        }
+
+        billingProject := ""
+
+        // err == nil indicates that the billing_project value was found
+        if bp, err := tpgresource.GetBillingProject(d, config); err == nil {
+                billingProject = bp
+        }
+
+        res, err := transport_tpg.SendRequest(transport_tpg.SendRequestOptions{
+                Config:    config,
+                Method:    "GET",
+                Project:   billingProject,
+                RawURL:    url,
+                UserAgent: userAgent,
+        })
+        fmt.Println(res, err, "Test print")
+        if err != nil {
+                return transport_tpg.HandleNotFoundError(err, d, fmt.Sprintf("ApphubDiscoveredService %q", d.Id()))
+        }
+        
+        if err := d.Set("discovered_service", flattenApphubDiscoveredService(res["discoveredService"], d, config)); err != nil {
+                return fmt.Errorf("Error setting discovered service: %s", err)
+        }
+        
+        d.SetId(time.Now().UTC().String())
```

Comment:
I think you want the id to be some combination of `project`, `location`, and `service_uri`.